### DISCUSSION TITLE
Formating numbers with datatables as a service

### DISCUSSION THREAD

**Discussion by EnriqueRBT** [yajra/laravel-datatables#2620] [Q&A]
I want to format the price to european currency. It,s easy with datatables helpers as explained in: https://datatables.net/manual/data/renderers#Built-in-helpers

For exmaple:

`                {data: 'cost', render:$.fn.dataTable.render.number('.',',','4','',' €'), name: 'cost'},
`

The problem is if I use datatables as a service.  I can`t find any documentation about how to achieve this. 

` protected function getColumns()

    {
        return [
            Column::make('cost')->title('Coste')
............`

Thanks for helping.

**Comment by yajra** [comment]
Try this:

```php
Column::make('cost')->title('Coste')->renderJs('number', '.',',','4','',' €')
```

Or

```php
Column::make('cost')->title('Coste')->renderRaw("$.fn.dataTable.render.number('.',',','4','',' €')")
```

**Comment by yajra** [comment]
Another newer way is via server-side formatters. 

```php
    public function dataTable($query)
    {
        return datatables()
            ->eloquent($query)
            ->formatColumn('amount', MoneyFormatter::class)
            ->setRowId('id');
    }

    protected function getColumns()
    {
        return [
            Column::make('id'),
            Column::formatted('amount'),
````

Formatter Class:

```php
namespace App\DataTables\Formatters;

use Yajra\DataTables\Contracts\Formatter;

class MoneyFormatter implements Formatter
{
    public function format($value, $row)
    {
        return number_format($value, 2);
    }
}
```

**Comment by EnriqueRBT** [comment]
Thank you very much, you are really kind, and the second message is really useful for me that i'm learning. I trully apreciate it.
